src/Umbraco.Web.Common/ModelsBuilder/InvalidatableRazorViewEngine.cs

```diff
@@ -0,0 +1,38 @@
+using System.Diagnostics;
+using System.Reflection;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Mvc.Razor;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Umbraco.Cms.Web.Common.ModelsBuilder
+{
+    internal class InvalidatableRazorViewEngine : RazorViewEngine
+    {
+        public InvalidatableRazorViewEngine(
+            IRazorPageFactoryProvider pageFactory,
+            IRazorPageActivator pageActivator,
+            HtmlEncoder htmlEncoder,
+            IOptions<RazorViewEngineOptions> optionsAccessor,
+            ILoggerFactory loggerFactory,
+            DiagnosticListener diagnosticListener)
+            : base(pageFactory, pageActivator, htmlEncoder, optionsAccessor, loggerFactory, diagnosticListener)
+        {
+        }
+
+        internal bool Invalidate()
+        {
+            const BindingFlags bindingFlags = BindingFlags.NonPublic | BindingFlags.Instance;
+            MethodInfo? cacheSetter = typeof(RazorViewEngine).GetProperty("ViewLookupCache", bindingFlags)?.GetSetMethod(true);
```

Comment:
ViewLookupCache is `protected internal` with a private setter, so there's an alternative here.